src/php/Compiler/Parser/ExpressionParser/AtomParser.php

```diff
@@ -16,6 +18,15 @@
 
 final class AtomParser
```

Comment:
What about adding unit tests for this class? Especially the behaviour that you introduced on `parseKeyword()` is super interesting and complex, and it would be good to have a suite of unit tests covering those possible scenarios. 
This way we could think about refactoring the `KEYWORD_REGEX` using aliases or whatever else. It would be also great as documentation for the complexity here.